worked_examples/geospatial/workshop_april_2024/Workshop_AprilMay2024_part2.ipynb

```diff
@@ -0,0 +1,948 @@
+    "\n",
+    "It contains two important classes:\n",
+    "\n",
+    "- ``AnnotationsLoader`` - This is used to load and review your annotations and to create datasets and dataloaders which are used to train your model.\n",
+    "- ``ClassifierContainer`` - This is used to set up your model, train/fine-tune it using your datasets and to infer labels on new datasets."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "d90aa0e9",
+   "metadata": {},
+   "source": [
+    "## Load annotations"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ea3c5860",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from mapreader import AnnotationsLoader"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "37fbf016",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "annotated_images = AnnotationsLoader()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "15fef8ac",
+   "metadata": {},
+   "source": [
+    "__**TODO**__: Load your annotations. They are saved in your ``\"./annotations/\"`` directory as a ``.csv`` file. You'll need to look in your files to see the exact file name."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a8829374",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# annotated_images.load()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e981c2b3",
+   "metadata": {},
+   "source": [
+    "Running ``annotated_images.labels_map`` will show you you the indexing of your labels. This is so they can be treated as numbers instead of strings in the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1366b88e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "annotated_images.labels_map"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "7ab44301",
+   "metadata": {},
+   "source": [
+    "### Review labels"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "4f883e9d",
+   "metadata": {},
+   "source": [
+    "Before training your model, you should check your annotations and ensure you are happy with your labels.\n",
+    "\n",
+    "This can be done using the ``.review_labels()`` method.\n",
+    "\n",
+    "For example, to re-label image with ``id: 5``, type \"5\" into the text box, press enter, then type the label (number) you'd like to re-label it as and press enter again."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "162a44c1",
+   "metadata": {},
+   "source": [
+    "> _**NOTE**_: type ``exit`` to quit!"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "f807fc40",
+   "metadata": {},
+   "source": [
+    "__**TODO**__: Review your annotations."
```

Comment:
I got in a mess here when I entered an invalid label. Easier to reproduce in person I think.

Reply:
this is a good point, will add some more info here

Reply:
- Edit text to say use label string
- Fix bug if you add a new label
- Make it clear you don't need to use the quote marks

Reply:
@thobson88 I made a ticket about the new label thing, have edited text in the notebook on the string vs number thing.